plugins/net_plugin/net_plugin.cpp

```diff
@@ -1456,17 +1457,21 @@ namespace eosio {
 
       }
 
+      org = 0;
       send_time();
    }
 
    // called from connection strand
    void connection::send_time() {
-      time_message xpkt;
-      xpkt.org = rec;
-      xpkt.rec = dst;
-      xpkt.xmt = get_time();
-      org = xpkt.xmt;
-      enqueue(xpkt);
+      if (org == 0) { // do not send if there is already a time loop in progress
+         time_message xpkt;
+         xpkt.org = rec;
```

Comment:
I don't understand why we don't do: `xpkt.org  = get_time();` here?

Reply:
It might be clearer to make it `xpkt.org = 0` which is what `rec` is here. The idea is this is the start of the ping.

Reply:
I am confused, why would 0 be the start time of the ping, and not the current time?

Reply:
Because of this: https://github.com/AntelopeIO/leap/pull/1243/files#diff-fee9397188f7093ef7c6c15c4b268a4866d1c8e9518697ca9edfd2ea9233e5fcL3299-L3301

Reply:
OK, but how do we know that `rec == 0` when we call `send_time()` [here](https://github.com/AntelopeIO/leap/blob/aa25f11201cb4350eca3396878fba2c469adafb6/plugins/net_plugin/net_plugin.cpp#L1460-L1461).

If the intent to have `xpkt.org = 0` in that send_time message, why use `rec`?

Reply:
I'm not quite sure why we use `rec` as well. Why not?

```
   void connection::send_time() {
      if (org == 0) { // do not send if there is already a time loop in progress
         time_message xpkt;
         xpkt.org = 0; // initiating a ping. Actual org time is in xpkt.xmt 
         xpkt.rec = 0;
         xpkt.xmt = get_time();
         org      = xpkt.xmt;
         peer_dlog(this, "send init time_message: ${t}", ("t", xpkt));
         enqueue(xpkt);
      }
   }
```

Reply:
and what about this for the other one:

```
   void connection::send_time(const time_message& msg) {
      assert(org != 0 && org == msg.xmt);
      time_message xpkt { msg.xmt, msg.dst, get_time(), 0 };
      peer_dlog( this, "send time_message: ${t}, org: ${o}", ("t", xpkt)("o", org) );
      enqueue(xpkt);
   }
```

Reply:
OK, I give up, the way it is coded is beyond my brain capacity to fully reason on it. If it was up to me, I'd completely rewrite it, but for now I'll just trust you and approve this PR.

Reply:
I agree it could be clearer. For example, `rec` isn't even needed as a member variable. I'll take a swing at simplifying it a bit.

Reply:
The code that I think is hard to reason with is `handle_message(const time_message&)`. if it was me I would have two fully distinct  cases, either we receive a ping message with `msg.org == 0` (initiated from peer), or with `msg.org != 0` (a response to out ping.